swap/src/protocol/alice/event_loop.rs

```diff
@@ -264,8 +291,19 @@ where
             swap_id,
         };
 
-        if let Err(error) = self.swap_sender.send(swap).await {
-            tracing::warn!(%swap_id, "Swap cannot be spawned: {}", error);
+        // TODO: Consider doing this in main and add another, separate channel for
+        //  resuming
+
+        // swaps save peer id so we can resume
+        match self.db.insert_peer_id(swap_id, bob_peer_id).await {
+            Ok(_) => {
+                if let Err(error) = self.swap_sender.send(swap).await {
+                    tracing::warn!(%swap_id, "Swap cannot be spawned: {}", error);
+                }
+            }
+            Err(error) => {
+                tracing::warn!(%swap_id, "Unable to save peer-id, swap cannot be spawned: {}", error);
+            }
```

Comment:
Because then we would save it again when we resume. We cannot distinguish if we start or resume a swap on the outside. We could say that is OK (just overwrite the peer-id upon resuming) or create a separate channel for resuming. thoughts?

Reply:
Fair enough. I guess the 2nd element could also be an `Option<PeerId>` or you model the two different payloads as an `enum`: `NewSwap`, `ResumeSwap` where only `NewSwap` has a field for the `PeerId`.

Reply:
Unfortunately this has implications on what we can do in the tests. If we handle saving the peer-id in main we cannot really test this properly. Given the current test setup this is hard to change - we could extract resume into a separate component that can be instantiated in both main and test, but I would see this in a follow up.

Reply:
After some more thinking the proper solution would be to extract building the swap to be done on the outside.

Rough outline of this solution:

- Have a separate component to load unfinished swaps upon startup.
- Have a swap builder on the outside (main / test-setup) that constructs the swap upon events from the event loop (execution setup finished) or upon loading unfinished swaps component
    - Send the peer-id back through the channel in case we construct a new swap.
    - Load the peer-id from the database in case of resume
- Allow constructing a handle through the event loop (peer id known by caller)
- Extract spawning the swap into dedicated function (to be used for new/resume - would be great if we can use main and test).

I will see to give this a try as a follow-up. I'd say we get this merged first so we don't waste time.